includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -73,16 +72,25 @@ public static function factory(
 	}
 
 	public function run() {
-		if ( !$this->user->isRegistered() ) {
+		// Rest Validator returns null if submitted as form
```

Comment:
Hm, are you sure? From looking at the code, it seems like `run()` should never even run if validation fails, at least when using MediaWiki’s `Router`.

Reply:
It exists before the body validator is run here https://github.com/wikimedia/mediawiki/blob/REL1_35/includes/Rest/Validator/Validator.php#L163

Reply:
:roll_eyes: I see, thanks.